packages/database/src/migrations/20210818014247-AddFijiCovid7DayPositivityIndicator-modifies-data.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+import {
+  insertObject,
+  generateId,
+  findSingleRecord,
+  findSingleRecordBySql,
+  deleteObject,
+} from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const INDICATOR_1 = {
+  code: 'COVID_FJ_7_Day_Rolling_Pos_Tests',
+  builder: 'analyticArithmetic',
+  config: {
+    formula: new Array(7)
+      .fill(0)
+      .map((_, i) => `PositiveTests${i + 1}Day${i === 0 ? '' : 's'}Ago`)
+      .join(' + '),
+    parameters: new Array(7).fill(0).map((_, i) => ({
+      // Note: from 0 to 6 days ago
+      code: `PositiveTests${i + 1}Day${i === 0 ? '' : 's'}Ago`,
+      config: {
+        formula: 'COVIDTest_FJPosTest',
+        aggregation: [
+          'FINAL_EACH_DAY',
+          {
+            type: 'OFFSET_PERIOD',
+            config: {
+              offset: i + 1,
+              periodType: 'day',
+            },
+          },
+        ],
+      },
+      builder: 'analyticArithmetic',
+    })),
+    aggregation: 'FINAL_EACH_DAY',
+    defaultValues: new Array(7)
+      .fill(0)
+      .map((_, i) => `PositiveTests${i + 1}Day${i === 0 ? '' : 's'}Ago`)
+      .reduce((acc, code) => ({ ...acc, [code]: 0 }), {}),
+  },
+};
+
+const INDICATOR_2 = {
+  code: 'COVID_FJ_7_Day_Rolling_Num_Tests',
+  builder: 'analyticArithmetic',
+  config: {
+    formula: new Array(7)
+      .fill(0)
+      .map((_, i) => `NumTests${i + 1}Day${i === 0 ? '' : 's'}Ago`)
+      .join(' + '),
+    parameters: new Array(7).fill(0).map((_, i) => ({
+      // Note: from 0 to 6 days ago
+      code: `NumTests${i + 1}Day${i === 0 ? '' : 's'}Ago`,
+      config: {
+        formula: 'COVIDTest_FJNumTest',
+        aggregation: [
+          'FINAL_EACH_DAY',
+          {
+            type: 'OFFSET_PERIOD',
+            config: {
+              offset: i + 1,
+              periodType: 'day',
+            },
+          },
+        ],
+      },
+      builder: 'analyticArithmetic',
+    })),
+    aggregation: 'FINAL_EACH_DAY',
+    defaultValues: new Array(7)
+      .fill(0)
+      .map((_, i) => `NumTests${i + 1}Day${i === 0 ? '' : 's'}Ago`)
+      .reduce((acc, code) => ({ ...acc, [code]: 0 }), {}),
+  },
+};
```

Comment:
Didn't these indicators already get added in another PR?  https://github.com/beyondessential/tupaia/pull/3209

Reply:
Yes... they did. My hope was that because they are the same commit they would just merge nicely? And if not that we could cross that bridge when we come to it. What should I do instead?

Reply:
Ummm if you squash n merged your other commit then I don't think it'll be a clean merge... might be if the files are identical I guess...

I'd usually recommend rebasing in this scenario, to clean up the commit history. Happy to give you a quick explanation if you like?

Otherwise, just test locally first to ensure the merge goes through cleanly?

Reply:
Sounds good, how about I rebase whichever one gets merged second, and test it locally as well?

Reply:
Sounds good to me! Also, might just mention that indicators can actually be created in the Admin panel, so if you'd wanted to do that it would've been fine. I prefer to do that when I'm sharing an indicator between multiple vizes 👍

Reply:
Ah, that sounds like it would have been the best way. I'll keep it in mind for next time!

Reply:
> might be if the files are identical I guess
For reference, it was a clean merge. Still not sure it was the best way to go commit-history-wise, but always good to know